chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -587,13 +587,34 @@
         mode 0644
         owner "root"
         group "root"
-        source "crowbar_join.ubuntu.sh.erb"
-        variables(admin_web: admin_web,
-                  os_codename: os_codename,
-                  crowbar_repo_web: crowbar_repo_web,
-                  admin_ip: admin_ip,
-                  provisioner_web: provisioner_web,
-                  web_path: web_path)
+        source "crowbar_join.sh.erb"
+        variables(admin_ip: admin_ip,
+                  web_port: web_port,
+                  ntp_servers_ips: ntp_servers_ips,
+                  platform_family: "debian",
+                  target_platform_version: target_platform_version)
+      end
+
+      if os == "debian-8"
+        param_install_url = params["install_url"]
+        debian_arch = case arch
+          when "x86_64"
+            "amd64"
+          when "ppc64le"
+            "ppc64el"
+          else
+            arch
+          end
+
+        kernel_path = Pathname.new(os_dir).join("install").join(kernel)
+        initrd_path = Pathname.new(os_dir).join("install").join(initrd)
+
+        missing_files = !kernel_path.exist? || !initrd_path.exist?
+
+        if missing_files && !param_install_url.nil? && !param_install_url.empty?
+          `wget --timeout 5 -O #{os_dir}/install/netboot.tar.gz #{param_install_url}/debian/dists/jessie/main/installer-#{debian_arch}/current/images/netboot/netboot.tar.gz && tar xf #{os_dir}/install/netboot.tar.gz -C #{os_dir}/install`
```

Comment:
Metrics/LineLength: Line is too long. [237/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)